### PR TITLE
Feature/ros1 ros2 comparison table

### PR DESCRIPTION
**Pull request by hello-atharva**
## Summary

This PR adds a feature comparison doc between our ROS offerings so that users can have a better understanding our of migration progress.

## How-to-test

--

### CODE REVIEW

File: ros2/feature_comparison.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+| Stretch MoveIt 2        | moveit_py                              | Multiplanning pipeline                               | ✕     | ✓     | In review                           |
+| Stretch MoveIt 2        | moveit_py                              | Octomap plugin                                       | ✕     | ✓     | In review                           |
+| Stretch MoveIt 2        | moveit                                 | End effector pose goals                              | ✕     | ✕     | ROS 2 pending                       |
+| Stretch MoveIt 2        | moveit                                 | Hybrid planning                                      | ✕     | ✕     | ROS 2 pending                       |
+| Web-based Teleoperation | web_interface                          | operator                                             | ✓     | ✕     | ROS 2 pending                       |
+| Web-based Teleoperation | web_interface                          | robot                                                | ✓     | ✕     | ROS 2 pending                       |
```
Comment (hello-chintan) [inline ros2/feature_comparison.md:114]: We don't ship this with the robots yet. I'd not add it to this list.

File: ros2/feature_comparison.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+| Stretch Install         | Humble                                 | Mesh and URDF files                                  | NA    | ✓     |                                     |
+| Stretch Install         | Iron                                   | Installation scripts                                 | NA    | ✓     |                                     |
+| Stretch Install         | Iron                                   | Mesh and URDF files                                  | NA    | ✓     |                                     |
+| Stretch Core            | stretch_driver                         | JointTrajectoryAction Server, Position mode          | ✓     | ✓     |                                     |
+| Stretch Core            | stretch_driver                         | JointTrajectoryAction Server, Navigation mode        | ✓     | ✓     |                                     |
+| Stretch Core            | stretch_driver                         | JointTrajectoryAction Server, Trajectory mode        | ✕     | ✓     | Smooth preemption missing           |
```
[the remark this answers is not in the document]
Comment (hello-chintan) [reply ros2/feature_comparison.md:13]: Meant the "smooth preemption missing" comment.

File: ros2/feature_comparison.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+| Stretch Core            | rplidar                                | Driver                                               | ✓     | ✓     |                                     |
+| Stretch Core            | rplidar                                | LaserScan filtering                                  | ✓     | ✓     |                                     |
+| Stretch Core            | d435i                                  | Driver                                               | ✓     | ✓     |                                     |
+| Stretch Core            | d435i                                  | High/Low resolution launch                           | ✓     | ✓     |                                     |
+| Stretch Core            | d435i                                  | IMU                                                  | ✓     | ✓     |                                     |
+| Stretch Core            | d435i                                  | Aligned depth                                        | ✓     | ✕     | ROS 2 pending                       |
```
[the remark this answers is not in the document]
Comment (hello-chintan) [reply ros2/feature_comparison.md:28]: Meant the Aligned depth line.

**Comment by hello-chintan** [comment]
Barring the above comments, looks good to merge! Thanks!

**Comment by hello-binit** [comment]
Merging the comparison table created in this PR, but it will need to be updated before it can be displayed on the docs site.

**Pull request merged**